### PR TITLE
pubsub aws snssqs update

### LINKED ISSUE #1899: SNS/SQS PubSub: missing sqsDeadLettersQueueName info
**What content needs to be created or modified?**

Support for SQS Dead Letter Queue was introduced in dapr 1.4 but the documentation is lacking that information in terms of "Spec metadata fields" missing the new `sqsDeadLettersQueueName` field and information about the dependency of setting this metadata field and `messageReceiveLimit`.

**Describe the solution you'd like**

- add description for the missing `sqsDeadLettersQueueName` metadata field spec
- add note about `messageReceiveLimit` that becomes mandatory if `sqsDeadLettersQueueName` is set
- add note about  `messageReceiveLimit` meaning in the context of specifying the `sqsDeadLettersQueueName` as it will be affecting the number of times a message is fetched before it is moved to the DeadLetters Queue, unlike when there is no DeadLetters Queue specified

**Where should the new material be placed?**

https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-aws-snssqs/

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

**Additional context**

### LINKED ISSUE #1977: AWS SNS/SQS support for FIFO - missing metadata fields
**What content needs to be created or modified?**

Related to https://github.com/dapr/components-contrib/issues/1205
Once the above and its accompanying PR are merged the below fields need to be added to the documentation

**Describe the solution you'd like**
add to the AWS SNS/SQS [Metadata information](https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-aws-snssqs/#spec-metadata-fields) the following setup fields:

```
field | required | details
-----------------------
fifo  | N | Boolean ("true" or "false") value enabling SNS/SQS FIFO support with automatic message deduplication
fifoMessageGroupID | N | Specify the FIFO publisher [Message Group ID](https://docs.aws.amazon.com/sns/latest/api/API_Publish.html). Default is auto creation of the Message Group ID per sidecar
```

**Where should the new material be placed?**

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

**Additional context**

### LINKED ISSUE #2035: PubSub AWS SnsSqs - disableEntityManagement feature documentation
**What content needs to be created or modified?**
new PubSub AWS SNS/SQS metadata field `disableEntityManagement` should be described and added

**Describe the solution you'd like**
`disableEntityManagement` should be added to the metadata fields table (see below)

**Where should the new material be placed?**
https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-aws-snssqs/#spec-metadata-fields

Essentially it has the same functionality as described in https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-azure-servicebus/#spec-metadata-fields thus copying of the same `disableEntityManagement` description can be done

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

**Additional context**

### LINKED ISSUE #2092: Pubsub AWS SNS/SQS - disableDeleteOnRetryLimit & assetsManagementTimeoutSeconds
**What content needs to be created or modified?**
Pubsub AWS SNS/SQS metadata setting for key: `disableDeleteOnRetryLimit` and `assetsManagementTimeoutSeconds` should be added

**Describe the solution you'd like**

**Where should the new material be placed?**

https://github.com/dapr/docs/blob/c7676b5be237388622fd1fa187fa08d88bd85af7/daprdocs/content/en/reference/components-reference/supported-pubsub/setup-aws-snssqs.md#spec-metadata-fields

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

https://github.com/dapr/components-contrib/pull/1432

**Additional context**

### PR DESCRIPTION
**Pull request by amimimor**
Signed-off-by: Amit Mor <amit.mor@hotmail.com>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

PubSub AWS SNS SQS documentation updates following current additions (already in "master" branch) to `components-contrib` of the aforementioned component.

I've followed current Azure/Pubsub/Servicebus "Component format" example that details all possible specifications while noting as a comment Optional specs.

I've also added basic security policies needed on AWS to use this component and the extended policies needed when using the new `disableEntityManagement` option

Included documentation of additional v1.6 features: `disableDeleteOnRetryLimit` and `assetsManagementTimeoutSeconds`

### CODE REVIEW

**Comment by amimimor** [comment]
reviewers - please visually render to make sure I haven't messed (mostly the main component table) things up as my editor an I had a fight over inserting spaces and column widths 🙏🏿

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-aws-snssqs.md [hunk 1]
```diff
@@ -46,15 +59,25 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
+| disableEntityManagement | N  | When set to true, SNS topics, SQS queues and the SQS subscriptions to SNS do not get created automatically. Default: `"false"` | `"true"`, `"false"`
+
+* Dapr created SNS topic and SQS queue names conform with [AWS specifications](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/quotas-queues.html). By default, Dapr creates an SQS queue name based on the consumer `app-id`, therefore Dapr might perform name standardization to meet with AWS specifications.
+* Using SQS FIFO (`fifo` metadata field set to `"true"`), per AWS specifications, provides message ordering and deduplication, but incurs a lower SQS processing throughput, among other caveats
+* Please be aware that specifying `fifoMessageGroupID` limits the number of concurrent consumers of the FIFO queue used to only one but guarantees global ordering of messages published by the app's Dapr sidecars. See [this](https://aws.amazon.com/blogs/compute/solving-complex-ordering-challenges-with-amazon-sqs-fifo-queues/) post to better understand the topic of Message Group IDs and FIFO queues.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-aws-snssqs.md]:
```suggestion
* Be aware that specifying `fifoMessageGroupID` limits the number of concurrent consumers of the FIFO queue used to only one, but guarantees global ordering of messages published by the app's Dapr sidecars. See [this](https://aws.amazon.com/blogs/compute/solving-complex-ordering-challenges-with-amazon-sqs-fifo-queues/) post to better understand the topic of Message Group IDs and FIFO queues.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-aws-snssqs.md [hunk 2]
```diff
@@ -133,8 +156,71 @@ spec:
...
+        "arn:aws:sqs:AWS_REGION:AWS_ACCOUNT_ID:*"
+      ]
+    }
+  ]
+}
+```
 Use the `AWS account ID` and `AWS account secret` and plug them into the `accessKey` and `secretKey` in the component metadata using Kubernetes secrets and `secretKeyRef`.
+
+
+Alternatively, if you want to provision the SNS and SQS assets using your own tool of choice (e.g. Terraform), while preventing Dapr from doing so dynamically, you will need to enable `disableEntityManagement` and assign your Dapr-using application with an IAM Role having a Policy such as:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-aws-snssqs.md]:
```suggestion
Alternatively, if you want to provision the SNS and SQS assets using your own tool of choice (e.g. Terraform), while preventing Dapr from doing so dynamically, you need to enable `disableEntityManagement` and assign your Dapr-using application with an IAM Role having a Policy such as:
```

**Review by msfussell — COMMENTED** [review]
Looks great! Did some light editing here. Please review

**Comment by amimimor** [comment]
@msfussell sorry for running over your comments here with a force push as I was "tricked" by github suggestions to accept your changes within the github-ui. That introduced un-"signed-off-by" commits and the force push was therefore mandatory.

**Comment by msfussell** [comment]
@amimimor  - One question. Does the Component format represent use AWS/SNS in production or some other scenario. Normally either we list all the metadata options or (less often) list this for a particular scenario (I notice that you do not have local endpoint list). If the latter may be worth adding a sentence before just saying the typically scenario to help people.

**Review by msfussell — APPROVED** [review]
LGTM

**Comment by msfussell** [comment]
Thanks for the contribution!

**Pull request merged**